src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/IAggregateValidator.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// Runs a set of <see cref="IValidator"/>s.
+    /// </summary>
+    public interface IAggregateValidator
+    {
+        /// <summary>
+        /// Human readable name of this <see cref="IAggregateValidator"/>.
+        /// </summary>
+        string Name { get; }
+
+        /// <summary>
+        /// Runs each <see cref="IValidator"/> and returns an <see cref="AggregateValidationResult"/> containing all results.
+        /// </summary>
+        Task<AggregateValidationResult> ValidateAsync(ValidationContext data);
+    }
+}
```

Comment:
Great observation--I noticed this myself and spent some time thinking about it--but I don't think it's feasible.

Basically, the problem is that `ValidationResult` represents a state and not a service. `AggregateAuditingService` is a service that says "I will call a list of other services". `AggregateValidationResult` represents a group of `ValidationResult`s. The distinction is that the individual methods of `AuditingService` still have meaning in the context of an `AggregateAuditingService`, but the individual properties of `ValidationResult` don't have meaning in the context of an `AggregateValidationResult`. In other words, it's the distinction between `string` and `IEnumerable<string>`.